array/DNAm/preprocessing/jobSubmission/1_runDNAmQC.sh

```diff
@@ -10,33 +10,35 @@
 #SBATCH --output=DNAm/logFiles/%u/QCDNAdata.log # output file
 #SBATCH --job-name=QCDNAdata
 
+#------------------------------------------------------
+
+# 1. the first input is your project folder.
+# 2. the second input is the full path to config file.
+# 3. the third input is a the path to the config.r file.
+
+#-----------------------------------------------------
+
+
 ## print start date and time
 echo Job started on:
 date -u
 JOBNAME="QCDNAdata"
-    
+
 ## needs to be executed from the scripts folder
 echo "Changing Folder to: " $SLURM_SUBMIT_DIR
 cd $SLURM_SUBMIT_DIR
 
 ## format paths in config file with project name
 echo "Loading config file for project: " $1
 export PROJECT=$1
-source ./DNAm/config/config.txt 
-
-## load R config file
-## default file used unless alternative provided on command line
-if [ -z "$2" ];
-then
-    RCONFIG=$SLURM_SUBMIT_DIR/DNAm/config/config.r
-else
-    RCONFIG=$2
-fi
+# the second input should be config file
+source $2 || exit 1
+RCONFIG=$3
```

Comment:
This is a much better way of doing it than before. now the config files (for the shell script and the Rscripts) can be placed anywhere. The user specifying the path means we don't need to force the script from being ran in a particular spot anymore.